python/TestHarness/TestHarness.py

```diff
@@ -956,3 +959,20 @@ def postRun(self, test, timing):
     # Insert the data into the database
     cr.executemany('insert into timing values (?,?,?,?,?,?,?)', data)
     con.commit()
+
+class TestHarnessTester(object):
+  """
+  Class for running TestHarness unit tests.
+  """
+  def __init__(self, argv, *args):
+    self._argv = argv
+
+  def findAndRunTests(self):
+    """
+    Execute the unittests for the TestHarness.
+    """
+    location = os.path.join(os.path.dirname(__file__), 'unit_tests')
+    loader = unittest.TestLoader()
+    suite = loader.discover(location)#, pattern)
+    runner = unittest.TextTestRunner(verbosity=2)
+    self.error_code = runner.run(suite).wasSuccessful()
```

Comment:
`error_code` should be an `int` rather than a `bool` like you have here.
Something like `self.error_code = int(not runner.run(suite).wasSuccessful())`